File: src/main/java/io/github/mjaroslav/bon2gradle/deobf/Bon2ProgressListenerErrorHandler.java
package io.github.mjaroslav.bon2gradle.deobf;

import io.github.mjaroslav.bon2.data.IErrorHandler;
import io.github.mjaroslav.bon2.data.IProgressListener;
import lombok.RequiredArgsConstructor;
import org.gradle.api.Project;
import org.jetbrains.annotations.NotNull;

// TODO: Make logs in BON2 fork more informative
@RequiredArgsConstructor
public class Bon2ProgressListenerErrorHandler implements IProgressListener, IErrorHandler {
    private final @NotNull Project project;

    @Override
    public void start(int max, String label) {
        project.getLogger().info(label);
    }

    @Override
    public void startWithoutProgress(String label) {
        project.getLogger().info(label);
    }

    @Override
    public void setProgress(int value) {
        // NO-OP
    }

    @Override
    public void setMax(int max) {
        // NO-OP
    }

    @Override
    public void setLabel(String label) {
        project.getLogger().info(label);
    }

    @Override
    public boolean handleError(String message, boolean warning) {
        if (warning) project.getLogger().warn(message);
        else project.getLogger().error(message);
        return true;
    }
}

File: src/main/kotlin/deobf/Bon2ProgressListenerErrorHandler.kt
package io.github.mjaroslav.bon2gradle.deobf

import io.github.mjaroslav.bon2.data.IErrorHandler
import io.github.mjaroslav.bon2.data.IProgressListener
import org.gradle.api.logging.Logger

class Bon2ProgressListenerErrorHandler constructor(private val logger: Logger) : IProgressListener, IErrorHandler {
    override fun start(max: Int, label: String?) {
        logger.info(label)
    }

    override fun startWithoutProgress(label: String?) {
        logger.info(label)
    }

    override fun setProgress(value: Int) {
        // NO-OP
    }

    override fun setMax(max: Int) {
        // NO-OP
    }

    override fun setLabel(label: String?) {
        logger.info(label)
    }

    override fun handleError(message: String?, warning: Boolean): Boolean {
        if (warning) logger.warn(message) else logger.error(message)
        return true
    }
}
